File: tracker/src/main/java/com/tracker/Entity/Note.java
package com.tracker.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import com.fasterxml.jackson.annotation.JsonBackReference;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "note")
public class Note {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String content;
    
    @ManyToOne(fetch = FetchType.LAZY)
    private Category category;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonBackReference("user-notes")
    private User user;

    private LocalDateTime createdAt = LocalDateTime.now();

    public Note() {}

    public Note(String title, String content, Category category, LocalDateTime createdAt) {
        this.title = title;
        this.content = content;
        this.category = category;
        this.createdAt = createdAt;
    }
}

File: frontend/src/types/Note.ts
/**
 * Note interface - соответствует структуре Java Entity/DTO от бэкенда
 * - id: Уникальный идентификатор
 * - title: Заголовок заметки
 * - content: Основной текст
 * - category: Категория заметки (объект Category)
 * - createdAt: Дата создания (ISO 8601 format)
 */

export interface Category {
  id: number;
  name: string;
}

export interface Note {
  id: number;
  title: string;
  content: string;
  category: Category;
  createdAt: string; // ISO 8601 date string, e.g., "2024-01-20T10:00:00"
}

export interface NoteRequest {
  title: string;
  content: string;
  categoryId?: number;
}